src/ucp/api/ucp.h

```diff
@@ -1283,6 +1341,62 @@ typedef struct {
 } ucp_request_param_t;
 
 
+/**
+ * @ingroup UCP_WORKER
+ * @brief Active Message handler parameters passed to
+ *        @ref ucp_worker_set_am_data_handler routine.
+ */
+typedef struct ucp_am_handler_param {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref ucp_am_handler_attr_field. Fields not specified in this mask will
+     * be ignored. Provides ABI compatibility with respect to adding new fields.
+     */
+    uint64_t                 field_mask;
+
+    /**
+     * Active Message id.
+     */
+    uint16_t                 id;
+
+    /**
+     * Active Message callback. To clear the already set callback, this value
+     * should be set to NULL.
+     */
+    ucp_am_data_callback_t   cb;
+
+    /**
+     * Active Message argument, which will be passed in to every invocation of
+     * @ref ucp_am_data_callback_t function as the @a arg argument.
+     */
+    void                     *arg;
+
+    /**
+     * Handler flags as defined by @ref ucp_am_cb_flags.
+     */
+    uint32_t                 flags;
+} ucp_am_handler_param_t;
+
+
+/**
+ * @ingroup UCP_WORKER
+ * @brief Operation parameters provided in @ref ucp_am_data_callback_t callback.
+ */
+struct ucp_am_recv_param {
+    /**
+     * Mask of valid fields in this structure and receive operation flags, using
```

Comment:
need to write this can also contain flags, not just valid fields

Reply:
it is already here - `and receive operation flags`

Reply:
ok